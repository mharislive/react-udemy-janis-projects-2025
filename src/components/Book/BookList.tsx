import Book from "../Book/Book";
import { BOOKS } from "./data";

const BookList = () => {
  return (
    <section className="grid grid-cols-1 md:grid-cols-3 gap-4 text-center">
      {BOOKS.map((book) => (
        <Book key={book.id} {...book} />
      ))}
    </section>
  );
};

export default BookList;
