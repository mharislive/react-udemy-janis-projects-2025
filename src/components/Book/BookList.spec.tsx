import { render, screen } from "@testing-library/react";
import { axe } from "vitest-axe";
import BookList from "./BookList";

describe("BookList", () => {
  it("should render successfully", () => {
    render(<BookList />);
  });

  it("should render 3 books", () => {
    render(<BookList />);
    expect(screen.getAllByRole("article")).toHaveLength(3);
  });

  it("should have no accessibility violations", async () => {
    const { container } = render(<BookList />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it("should match the snapshot", () => {
    const { asFragment } = render(<BookList />);
    expect(asFragment()).toMatchSnapshot();
  });
});
