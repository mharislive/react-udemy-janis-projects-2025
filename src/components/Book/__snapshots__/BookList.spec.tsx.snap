// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BookList > should match the snapshot 1`] = `
<DocumentFragment>
  <section
    class="grid grid-cols-1 md:grid-cols-3 gap-4 text-center"
  >
    <article
      class="p-5 rounded-lg border-1"
    >
      <img
        alt="Atomic Habits"
        class="mx-auto mb-4 max-w-full"
        src="https://m.media-amazon.com/images/I/81F90H7hnML._SY425_.jpg"
      />
      <h2
        class="text-2xl mb-1"
      >
        Atomic Habits
      </h2>
      <h3
        class="text-sm"
      >
        By James Clear
      </h3>
    </article>
    <article
      class="p-5 rounded-lg border-1"
    >
      <img
        alt="Word Power Made Easy"
        class="mx-auto mb-4 max-w-full"
        src="https://m.media-amazon.com/images/I/71DortY2MXL._SY425_.jpg"
      />
      <h2
        class="text-2xl mb-1"
      >
        Word Power Made Easy
      </h2>
      <h3
        class="text-sm"
      >
        By Norman Lewis
      </h3>
    </article>
    <article
      class="p-5 rounded-lg border-1"
    >
      <img
        alt="The Psychology of Money Paperback"
        class="mx-auto mb-4 max-w-full"
        src="https://m.media-amazon.com/images/I/71XEsXS5RlL._SY425_.jpg"
      />
      <h2
        class="text-2xl mb-1"
      >
        The Psychology of Money Paperback
      </h2>
      <h3
        class="text-sm"
      >
        By Morgan Housel
      </h3>
    </article>
  </section>
</DocumentFragment>
`;
